packages/vow/test/test-watch.js

```diff
@@ -50,6 +68,28 @@ const runTests = async t => {
   resolver3.reject(Error('disco2'));
   resolver3.resolve(vow2);
   t.is(await when(watch(connVow3P, makeAckWatcher(packet))), 'rejected');
-};
+});
+
+test('promise watcher', async t => {
+  const zone = makeHeapZone();
+  const { watch, when } = prepareVowTools(zone, {
+    rejectionMeansRetry: reason => reason === 'disconnected',
+  });
+
+  const pk = makePromiseKit();
+  pk.resolver.reject('disconnected');
+
+  const vow = watch(pk.promise, {
+    onFulfilled(value) {
+      t.log(`onfulfilled ${value}`);
+      t.fail('should not fulfil');
+      return 'fulfilled';
+    },
+    onRejected(reason) {
+      t.is(reason, 'disconnected');
+      return `rejected ${reason}`;
+    },
+  });
 
-test('ack watcher - shim', runTests);
+  t.is(await when(vow), 'rejected disconnected');
```

Comment:
What is this actually testing? I think we need some comments explaining why e.g. `rejectionMeansRetry: reason => reason === 'disconnected'` is not expected to have an effect here.

Reply:
I renamed the test and added some comments.